### DISCUSSION TITLE
Cannot use i3bar or polybar as tray (XFCE + i3 problem)

### DISCUSSION THREAD

**Discussion by osa1** [i3/i3#4773] [Q&A]
I asked this in XFCE forums (https://forum.xfce.org/viewtopic.php?pid=65504) but one of the devs think this may be an issue with i3, so here it goes.

I've been using XFCE with i3 for many years. Recently I reinstalled my Xubuntu 18.04 + i3 system to Xubuntu 21.10 (not an upgrade! clean install). I installed i3 from the package manager, I disabled XFCE desktop processes. The problem is i3bar is not used as system tray so I don't see system tray icons anymore. If I run `xfce4-panel --disable-wm-check`, tray icons appear there, but not in i3bar. I also tried installing polybar, but no luck.

One interesting note here is polybar can detect when another process is acting as system tray and prints a warning. It does not print that warning, but it also does not show the tray icons.

XFCE processes that I'm running: (generated by searching for "xfce" in htop)

- xfce4-session
- xfce4-screensaver
- xfce4-power-manager
- xfce4-notifyd
- xfce4-appfinder

Any ideas on what may be the problem, and how to fix this?

Thanks

**Comment by Airblader** [comment]
> one of the devs think this may be an issue with i3

I disagree with that suspicion since the window manager isn't involved at all in the system tray protocol. There's a host of which there can only be one, but that host would be i3bar or polybar, both of which are separate processes.

What tray applications are you running? I'd say you can figure out who owns the tray selection, but if polybar has a check for that I assume that is what it's doing for you already.

**Reply by osa1** [reply]
Hi @Airblader, thanks for the response.

> What tray applications are you running?

By default only `NetworkManager`, but for testing I also run other processes that normally use tray, like OBS Studio or VLC. None of these apps appear in i3bar or polybar, but they appear in xfce4-panel if I run it with `xfce4-panel --disable-wm-check`.

**Reply by osa1** [reply]
> I'd say you can figure out who owns the tray selection

How do I do that?

**Reply by Airblader** [reply]
I think best would be getting a debug log from i3bar, see the very bottom of https://i3wm.org/docs/debugging.html.
